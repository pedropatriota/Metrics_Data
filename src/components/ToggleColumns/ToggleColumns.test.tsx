import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import { ToggleColumns } from ".";

const columns = {
  label: true,
  value: true,
  type: true,
  description: true,
  category: true,
};

describe("Snapshot test", () => {
  it("Should  represent the component", () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const toggleColumns = render(
      <ToggleColumns
        columns={columns}
        onCheckboxClick={() => console.log("clicked")}
      />
    );

    expect(toggleColumns).toMatchSnapshot();
  });
});
