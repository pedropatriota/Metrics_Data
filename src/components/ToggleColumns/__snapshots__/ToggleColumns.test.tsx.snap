// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Should  represent the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL bSAafL"
      >
        <div
          class="sc-gsDKAQ eTdEsu"
        >
          <label
            for="label"
          >
            label
          </label>
          <input
            checked=""
            id="label"
            name="label"
            type="checkbox"
          />
        </div>
        <div
          class="sc-gsDKAQ eTdEsu"
        >
          <label
            for="value"
          >
            value
          </label>
          <input
            checked=""
            id="value"
            name="value"
            type="checkbox"
          />
        </div>
        <div
          class="sc-gsDKAQ eTdEsu"
        >
          <label
            for="type"
          >
            type
          </label>
          <input
            checked=""
            id="type"
            name="type"
            type="checkbox"
          />
        </div>
        <div
          class="sc-gsDKAQ eTdEsu"
        >
          <label
            for="description"
          >
            description
          </label>
          <input
            checked=""
            id="description"
            name="description"
            type="checkbox"
          />
        </div>
        <div
          class="sc-gsDKAQ eTdEsu"
        >
          <label
            for="category"
          >
            category
          </label>
          <input
            checked=""
            id="category"
            name="category"
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL bSAafL"
    >
      <div
        class="sc-gsDKAQ eTdEsu"
      >
        <label
          for="label"
        >
          label
        </label>
        <input
          checked=""
          id="label"
          name="label"
          type="checkbox"
        />
      </div>
      <div
        class="sc-gsDKAQ eTdEsu"
      >
        <label
          for="value"
        >
          value
        </label>
        <input
          checked=""
          id="value"
          name="value"
          type="checkbox"
        />
      </div>
      <div
        class="sc-gsDKAQ eTdEsu"
      >
        <label
          for="type"
        >
          type
        </label>
        <input
          checked=""
          id="type"
          name="type"
          type="checkbox"
        />
      </div>
      <div
        class="sc-gsDKAQ eTdEsu"
      >
        <label
          for="description"
        >
          description
        </label>
        <input
          checked=""
          id="description"
          name="description"
          type="checkbox"
        />
      </div>
      <div
        class="sc-gsDKAQ eTdEsu"
      >
        <label
          for="category"
        >
          category
        </label>
        <input
          checked=""
          id="category"
          name="category"
          type="checkbox"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
